zmon-controller-app/src/main/java/org/zalando/zmon/config/OAuth2TokenExtractor.java

```diff
@@ -0,0 +1,20 @@
+package org.zalando.zmon.config;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+public class OAuth2TokenExtractor implements Function<Optional<String>, Optional<String>> {
+
+    @Override
+    public Optional<String> apply(Optional<String> t) {
+        return t.flatMap(header -> extractTokenFromHeader(header));
+    }
+
+    protected Optional<String> extractTokenFromHeader(String header) {
+        try {
+            return Optional.ofNullable(header.substring(7)); // Bearer
```

Comment:
Nice, I think we had that before but I'm not able to remember what the reason was to use '7' instead.